router/delivery/http/router_handler.go

```diff
@@ -105,7 +105,9 @@ func (a *RouterHandler) GetOptimalQuote(c echo.Context) (err error) {
 		tokenIn, tokenOutDenom = req.TokenOut, req.TokenInDenom
 	}
 
-	chainDenoms, err := mvc.ValidateChainDenomsQueryParam(c, a.TUsecase, []string{tokenIn.Denom, tokenOutDenom})
+	// We do not allow unlisted denoms since this is a production from the swap tool.
+	doesAllowUnlistedDenoms := false
```

Comment:
Should we allow client to specify `doesAllowUnlistedDenoms` param? Would defining `AllowUnlistedDenoms` field in `types.GetQuoteRequest` and setting default ( during Unmarsal ) make more sense and would allow to keep handlers smaller?

Reply:
Hmm not sure - we don't have a product use case for this.